crates/ruff/src/rules/tryceratops/rules/raise_within_try.rs

```diff
@@ -77,15 +82,55 @@ pub(crate) fn raise_within_try(checker: &mut Checker, body: &[Stmt], handlers: &
         return;
     }
 
+    // The names of exceptions handled by the `try` Stmt. We can't compare `Expr`'s since they have
+    // different ranges, but by virtue of this function's call path we know that the `raise`
+    // statements will always be within the surrounding `try`.
+    let handler_ids: FxHashSet<&str> = helpers::extract_handled_exceptions(handlers)
+        .into_iter()
+        .filter_map(|handler| {
+            if let Expr::Name(ast::ExprName { id, .. }) = handler {
+                Some(id.as_str())
+            } else {
+                None
+            }
+        })
+        .collect();
+
     let raises = {
         let mut visitor = RaiseStatementVisitor::default();
         visitor.visit_body(body);
         visitor.raises
     };
 
     for stmt in raises {
-        checker
-            .diagnostics
-            .push(Diagnostic::new(RaiseWithinTry, stmt.range()));
+        let Stmt::Raise(ast::StmtRaise { exc, .. }) = stmt else {
+            continue;
+        };
+
+        let Some(exception) = exc else {
+            continue;
+        };
+
+        let exc_name = get_function_name(exception.as_ref()).unwrap_or_default();
+        // We can't check exception sub-classes without a type-checker implementation, so let's
+        // just catch the blanket `Exception` for now.
+        if handler_ids.contains(exc_name) || handler_ids.contains("Exception") {
```

Comment:
Do we need to check if `Exception` is a builtin type?

Reply:
Do we have a way to do that? I thought it'd suffice to just go off of name, but if we have a more robust way happy to change it

Reply:
@MichaReiser just in case this slipped through

Reply:
Thanks for the pining. I, indeed, missed this message. 

Can we use [is_builtin](https://github.com/astral-sh/ruff/blob/7ac9e0252e3c278078e19a7053671fdc845986af/crates/ruff_python_semantic/src/model.rs#L239-L241) for this?